src/main/scala/zio/kafka/consumer/internal/Runloop.scala

```diff
@@ -218,9 +219,17 @@ private[consumer] final class Runloop(
             reqRecs.toArray[ByteArrayConsumerRecord](Array.ofDim[ByteArrayConsumerRecord](reqRecs.size))
           )
 
-        fulfillAction = fulfillAction *> req.cont.succeed(
-          concatenatedChunk.map(CommittableRecord(_, commit(_), Try(consumer.consumer.groupMetadata()).toOption))
-        )
+        fulfillAction = fulfillAction *> req.cont.succeed(concatenatedChunk.map { record =>
+          CommittableRecord(
+            record = record,
+            commitHandle = commit,
+            consumerGroupMetadata =
+              if (hasGroupId)
+                try Some(consumer.consumer.groupMetadata())
+                catch { case _: Throwable => None }
```

Comment:
Catching everything is not recommended
```suggestion
                catch { case NonFatal(_) => None }
```

Or simply: ```Try(consumer.consumer.groupMetadata()).toOption```

Reply:
Well catch thanks!

Reply:
`Try(..)` will allocate a try object to discard it right away, minimal optim but better not doing that somewhere we might pass a lot of time.